pkg/backup/client.go

```diff
@@ -71,31 +72,34 @@ func (bc *Client) GetTS(ctx context.Context, timeAgo string) (uint64, error) {
 	if err != nil {
 		return 0, errors.Trace(err)
 	}
+	backupTS := oracle.ComposeTS(p, l)
 
 	if timeAgo != "" {
 		duration, err := time.ParseDuration(timeAgo)
 		if err != nil {
 			return 0, errors.Trace(err)
 		}
-		t := duration.Nanoseconds() / int64(time.Millisecond)
-		log.Info("backup time ago", zap.Int64("MillisecondsAgo", t))
-
-		// check backup time do not exceed GCSafePoint
-		safePoint, err := GetGCSafePoint(ctx, bc.mgr.GetPDClient())
-		if err != nil {
-			return 0, errors.Trace(err)
+		if int(duration) <= 0 {
+			return 0, errors.New("negative timeago is allowed")
 		}
-		if p-t < safePoint.Physical {
-			return 0, errors.New("given backup time exceed GCSafePoint")
+		log.Info("backup time ago", zap.Duration("timeago", duration))
+
+		backupTime := oracle.GetTimeFromTS(backupTS)
+		backupAgo := backupTime.Add(-duration)
+		if backupTS < oracle.ComposeTS(oracle.GetPhysical(backupAgo), l) {
+			return 0, errors.New("backup ts overflow please choose a smaller timeago")
 		}
-		p -= t
+		backupTS = oracle.ComposeTS(oracle.GetPhysical(backupAgo), l)
 	}
 
-	ts := utils.Timestamp{
-		Physical: p,
-		Logical:  l,
+	// check backup time do not exceed GCSafePoint
+	safePoint, err := GetGCSafePoint(ctx, bc.mgr.GetPDClient())
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	if backupTS < safePoint {
```

Comment:
```suggestion
	if backupTS <= safePoint {
```

Just for consistency, since `pkg/backup/safe_point.go` uses `<=` 🙃 (BTW should it reuse `CheckGCSafepoint`? the only difference is the handing of `GetGCSafePoint` error.)